spinalcordtoolbox/scripts/sct_analyze_lesion.py

```diff
@@ -311,6 +325,210 @@ def _measure_axial_damage_ratio(self, im_data, p_lst, idx):
+            # Iterate across axial slices to compute tissue bridges
+            for axial_slice in axial_lesion_slices:
+                # Get the lesion segmentation mask of the selected 2D axial slice
+                slice_lesion_data = im_lesion_data[sagittal_slice, :, axial_slice]
+                # Get the spinal cord segmentation mask of the selected 2D axial slice
+                slice_sc_data = im_sc_data[sagittal_slice, :, axial_slice]
+                # Get the indices of the lesion mask for the selected axial slice
+                lesion_indices = np.where(slice_lesion_data)[0]
+                # Get the indices of the spinal cord mask for the selected axial slice
+                sc_indices = np.where(slice_sc_data)[0]
+
+                # Compute ventral and dorsal tissue bridges
+                dorsal_bridge_width = lesion_indices[0] - sc_indices[0]         # [0] returns the most dorsal elements
+                # if the lesion extends the spinal cord, the dorsal bridge is set to 0
+                if dorsal_bridge_width < 0:
+                    dorsal_bridge_width = 0
+                ventral_bridge_width = sc_indices[-1] - lesion_indices[-1]      # [-1] returns the most ventral elements
+                # if the lesion extends the spinal cord, the ventral bridge is set to 0
+                if ventral_bridge_width < 0:
+                    ventral_bridge_width = 0
+
+                tissue_bridges_dict[sagittal_slice, axial_slice] = \
+                    {'dorsal_bridge_width': dorsal_bridge_width,
+                     'ventral_bridge_width': ventral_bridge_width,
+                     'lesion_indices': lesion_indices}       # we store lesion indices for plotting
+
+        # --------------------------------------
+        # Get minimal tissue bridges
+        # --------------------------------------
+        # Convert the dictionary to a DataFrame (for easier manipulation)
+        # 1. Create a MultiIndex from the dictionary keys
+        index = pd.MultiIndex.from_tuples(tissue_bridges_dict.keys(), names=['sagittal_slice', 'axial_slice'])
+        # 2. Create the DataFrame using the MultiIndex and the dictionary values
+        tissue_bridges_df = pd.DataFrame(list(tissue_bridges_dict.values()), index=index)
+        # 3. Reset the index to make 'sagittal_slice' and 'axial_slice' as columns
+        tissue_bridges_df.reset_index(inplace=True)
+
+        # Initialize auxiliary dict to store info for plotting
+        plot_dict = {}
+        # Get slices of minimum dorsal and ventral tissue bridges for each sagittal slice
+        # NOTE: we get minimum because tissue bridges are quantified as the width of spared tissue at the minimum
+        # distance from cerebrospinal fluid to the lesion boundary
+        for sagittal_slice in sagittal_lesion_slices:
+            # Get df for the selected sagittal slice
+            df_temp = tissue_bridges_df[tissue_bridges_df['sagittal_slice'] == sagittal_slice]
```

Comment:
Very good suggestion! Thank you!
Code updated in https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/4489/commits/3d1da0afe9b42c31600d4a27a96b47275d70521f.

Reply:
Hmm, looking deeper, this change introduced issues with `axial slices`:

<details><summary>before 3d1da0a </summary>

<img width="500" alt="Screenshot 2024-06-17 at 4 57 09 PM" src="https://github.com/spinalcordtoolbox/spinalcordtoolbox/assets/39456460/6ee4defc-f034-41c8-be17-b9954609bfd7">

</details>

<details><summary>after 3d1da0a </summary>

<img width="500" alt="Screenshot 2024-06-17 at 4 57 40 PM" src="https://github.com/spinalcordtoolbox/spinalcordtoolbox/assets/39456460/56edf0cb-2bdd-4c52-91a7-4d0f2838eae6">

</details>

This can be easily solved by adding `inplace=True` to `df_temp.set_index('axial_slice', inplace=True)`. But then, there are issues with `['axial_slice']` indexing at the following lines:

https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/3d1da0afe9b42c31600d4a27a96b47275d70521f/spinalcordtoolbox/scripts/sct_analyze_lesion.py#L509
https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/3d1da0afe9b42c31600d4a27a96b47275d70521f/spinalcordtoolbox/scripts/sct_analyze_lesion.py#L517

Reply:
Since I wanted to continue working on [suggestions from collaborators](https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/4489#issuecomment-2168479780), I am thinking about reverting https://github.com/spinalcordtoolbox/spinalcordtoolbox/commit/3d1da0afe9b42c31600d4a27a96b47275d70521f